test/integration_test/basic_test.go

```diff
@@ -308,31 +321,128 @@ func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
+		}
+
+		logrus.Info("Delete stork pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork", cluster.Namespace, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
 		require.NoError(t, err)
 
-		logrus.Info("Enable CSI and validate StorageCluster")
+		logrus.Info("Delete stork-scheduler pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork-scheduler", cluster.Namespace, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
+		require.NoError(t, err)
+
+		logrus.Info("Enable Stork webhook-controller and validate StorageCluster")
 		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		require.NoError(t, err)
-		cluster.Spec.FeatureGates = map[string]string{"CSI": "true"}
-		cluster, err = ci_utils.UpdateStorageCluster(cluster)
+		// At this point this map should be <nil>
+		if len(cluster.Spec.Stork.Args) == 0 {
+			cluster.Spec.Stork.Args = make(map[string]string)
+		}
+		cluster.Spec.Stork.Args["webhook-controller"] = "true"
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		logrus.Info("Disable Stork webhook-controller and validate StorageCluster")
+		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		require.NoError(t, err)
+		cluster.Spec.Stork.Args["webhook-controller"] = "false"
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
 
-		// Sleep for 20 seconds to let operator start the update process
-		logrus.Debug("Sleeping for 20 seconds...")
-		time.Sleep(20 * time.Second)
+		logrus.Info("Remove Stork webhook-controller and validate StorageCluster")
+		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+		require.NoError(t, err)
+		delete(cluster.Spec.Stork.Args, "webhook-controller")
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		logrus.Info("Enable Stork hostNetwork and validate StorageCluster")
+		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+		require.NoError(t, err)
+		hostNetworkValue := true
+		cluster.Spec.Stork.HostNetwork = &hostNetworkValue
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		logrus.Info("Disable Stork hostNetwork and validate StorageCluster")
+		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+		require.NoError(t, err)
+		*cluster.Spec.Stork.HostNetwork = false
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
```

Comment:
return the cluster for this function so we don't need to get it again underneath.

Reply:
Added getting cluster at the end of UpdateAndValidateStorageCluster() and returning new liveCluster, it seems that UpdateStorageCluster() function doesn't return the very last recent version of StorageCluster object

Reply:
Fixed this in latest commit